### PR TITLE
[feat]tools-v2: add update copyset availflag

### PR DESCRIPTION
**Pull request by baytan0720**
### What problem does this PR solve?

Issue Number: #2357  

Problem Summary: support update availflag command in [curve tool](https://github.com/opencurve/curve/tree/master/tools-v2)

### What is changed and how it works?

What's Changed:
add tools-v2/pkg/cli/command/curvebs/list/unavailcopysets/unavailcopysets.go
add tools-v2/pkg/cli/command/curvebs/query/chunk/chunkinfo.go
add tools-v2/pkg/cli/command/curvebs/list/chunkserver/copyset_by_host.go
add tools-v2/pkg/cli/command/curvebs/set/set.go
add tools-v2/pkg/cli/command/curvebs/update/copyset/copyset.go
add tools-v2/pkg/cli/command/curvebs/update/copyset/availflag/availflag.go
modify tools-v2/internal/error/error.go
modify tools-v2/internal/utils/row.go
modify tools-v2/pkg/cli/command/curvebs/bs.go
modify tools-v2/pkg/config/bs.go
modify tools-v2/pkg/cli/command/curvebs/update/update.go
modify tools-v2/pkg/cli/command/curvebs/check/copyset/copyset.go

How it Works:
```
root@0ad49a8af612:/# curve bs update copyset-availflag -h               
Usage:  curve bs update copyset-availflag [flags]

set copyset availflag

Flags:
      --availflag             copysets available flag[required]
  -c, --conf string           config file (default is
                              $HOME/.curve/curve.yaml or
                              /etc/curve/curve.yaml)
      --dryrun                when dry run set true, no changes will be
                              made (default true)
  -f, --format string         output format (json|plain) (default "plain")
  -h, --help                  print help
      --mdsaddr string        mds address, should be like
                              127.0.0.1:6700,127.0.0.1:6701,127.0.0.1:6702
      --rpcretrytimes int32   rpc retry times (default 1)
      --rpctimeout duration   rpc timeout (default 10s)
      --showerror             display all errors in command
      --verbose               show some log

Examples:
$ curve bs update copyset-availflag --availflag=true --dryrun=false
```

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: tools-v2/internal/error/error.go [hunk 1]
```diff
@@ -420,7 +420,18 @@ var (
 	ErrBsOpNameNotSupport = func() *CmdError {
 		return NewInternalCmdError(53, "not support op[%s], only support: operator, change_peer, add_peer, remove_peer, transfer_leader")
 	}
-
+	ErrBsGetChunkServerInCluster = func() *CmdError {
+		return NewInternalCmdError(53, "get chunkserver in cluster fail, err: %s")
```
Comment (aspirer) [inline tools-v2/internal/error/error.go:424]: the errorno 53 is already used above, we should start with errorno 54.

File: tools-v2/pkg/cli/command/curvebs/update/copyset/availflag/availflag.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/common"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology/statuscode"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+const (
+	copysetAvailflagExample = `$ curve bs update copyset availflag --availflag=true --dryrun=false`
```
Comment (aspirer) [inline tools-v2/pkg/cli/command/curvebs/update/copyset/availflag/availflag.go:50]: `set` or `update`?
you use `set` in conversation, I think it should be `update`, we should correct it.

**Review by aspirer — CHANGES_REQUESTED** [review]
we also need to update doc in `tools-v2/README.md`.

you can refer to pr #2465

**Comment by baytan0720** [comment]
> we also need to update doc in `tools-v2/README.md`.
> 
> you can refer to pr #2465

Ok, I will fix all. This pr is not ready yet. I've been a little busy lately.

File: tools-v2/pkg/config/bs.go [hunk 3]
```diff
@@ -36,70 +36,79 @@ import (
...
+	VIPER_CURVEBS_SNAPSHOTDUMMYADDR   = "curvebs.snapshotDummyAddr"
+	CURVEBS_DRYRUN                    = "dryrun"
+	VIPER_CURVEBS_DRYRUN              = "curvebs.dryrun"
+	CURVEBS_DEFAULT_DRYRUN            = true
+	CURVEBS_AVAILFLAG                 = "availflag"
+	VIPER_CURVEBS_AVAILFLAG           = "curvebs.availflag"
+	CURVEBS_CHUNK_ID                  = "chunkid"
+	VIPER_CURVEBS_CHUNK_ID            = "curvebs.chunkid"
+	CURVEBS_CHUNKSERVER_ADDRESS       = "csaddress"
+	VIPER_CURVEBS_CHUNKSERVER_ADDRESS = "curvebs.csaddress"
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/config/bs.go:111]:
```suggestion
	CURVEBS_CHUNKSERVER_ADDRESS       = "chunkserveraddr"
	VIPER_CURVEBS_CHUNKSERVER_ADDRESS = "curvebs.chunkserverAddr"
```

File: tools-v2/pkg/cli/command/curvebs/list/chunkserver/chunkserver.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+func (gRpc *GetChunkServerInClusterRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return gRpc.mdsClient.GetChunkServerInCluster(ctx, gRpc.Request)
+}
+
+type ChunkserverCommand struct {
+	basecmd.FinalCurveCmd
+	Rpc      *GetChunkServerInClusterRpc
+	response *topology.GetChunkServerInClusterResponse
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/list/chunkserver/chunkserver.go:56]:
```suggestion
	ChunkServerInfos []*topology.ChunkServerInfo
```
is better?

File: tools-v2/pkg/cli/command/curvebs/list/unavailcopysets/unavailcopysets.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+func (lRpc *ListUnAvailCopySets) Stub_Func(ctx context.Context) (interface{}, error) {
+	return lRpc.topologyClient.ListUnAvailCopySets(ctx, lRpc.Request)
+}
+
+type UnAvailCopySetsCommand struct {
+	basecmd.FinalCurveCmd
+	Rpc      *ListUnAvailCopySets
+	response *topology.ListUnAvailCopySetsResponse
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/list/unavailcopysets/unavailcopysets.go:58]: ditto

File: tools-v2/pkg/cli/command/curvebs/query/chunk/chunkinfo.go [hunk 6]
```diff
@@ -0,0 +1,167 @@
+/*
+ *  Copyright (c) 2022 NetEase Inc.
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/query/chunk/chunkinfo.go:2]:
```suggestion
 *  Copyright (c) 2023 NetEase Inc.
```

File: tools-v2/pkg/cli/command/curvebs/list/chunkserver/chunkserver.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
+/*
+ *  Copyright (c) 2022 NetEase Inc.
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/list/chunkserver/chunkserver.go:2]:
```suggestion
 *  Copyright (c) 2023 NetEase Inc.
```

File: tools-v2/pkg/cli/command/curvebs/query/chunk/chunkinfo.go [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+		return fmt.Errorf("parse chunkid %v fail", chunkidList)
+	}
+	for i := 0; i < len(copysetIds); i++ {
+		rpc := &GetChunkInfoRpc{
+			Request: &chunk.GetChunkInfoRequest{
+				LogicPoolId: &logicalpoolIds[i],
+				CopysetId:   &copysetIds[i],
+				ChunkId:     &chunkIds[i],
+			},
+			Info: &basecmd.Rpc{[]string{addressList[i]}, timeout, retrytimes, "GetChunkInfo"},
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/query/chunk/chunkinfo.go:104]:
```suggestion
			Info: &basecmd.NewRpc{[]string{addressList[i]}, timeout, retrytimes, "GetChunkInfo"},
```

File: tools-v2/pkg/cli/command/curvebs/update/copyset/availflag/availflag.go [hunk 9]
```diff
@@ -0,0 +1,259 @@
+/*
+ *  Copyright (c) 2022 NetEase Inc.
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/update/copyset/availflag/availflag.go:2]:
```suggestion
 *  Copyright (c) 2023 NetEase Inc.
```

File: tools-v2/pkg/cli/command/curvebs/update/copyset/availflag/availflag.go [hunk 10]
```diff
@@ -0,0 +1,259 @@
+/*
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/update/copyset/availflag/availflag.go:1]: add use in readme

**Comment by baytan0720** [comment]
cicheck

**Comment by baytan0720** [comment]
cicheck

**Review by Cyber-SiKu — APPROVED** [review]

**Review by SeanHai — APPROVED** [review]

**Pull request merged**